### DISCUSSION TITLE
Multi-Select Picker and Arrays with Values Containing Commas in SQL

### DISCUSSION THREAD

**Discussion by Marcello1099** [Budibase/budibase#11599] [Help] [answered]
Hello,

I am using the multi-select picker to chose multiple options and insert this into the `WHERE` statement of my query. For example:
```
...
WHERE field = ANY(string_to_array({{ multi-select field }}, ','))
...
```

Now the issue with this approach is that my values sometimes include a comma; thus, making more values than actual and not getting a match. For instance:

```
'Yes, I like dogs' -> ['Yes', 'I like dogs'] and I would want it to be ['Yes, I like dogs' ]
```

Therefore, I wonder if there is any way to change the separator from the multi-picker or if there is a better approach to insert it into the SQL query. 

Thanks in advance!

**Comment by andz-bb** [comment, marked as answer]
hey @Marcello1099 

The multi-select picker's field value is an array so if you need to, you can use a JavaScript binding to join the array with a different delimiter before passing it to your query as a binding. Here's a quick example:

I want to be able to filter my products by brand, and one of those brands is stored in my db as `Dell, Alienware`.

<img width="829" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/74b3aefe-3e60-4b36-893f-bd4f18fca682">

I have my Postgres query set up with a `brands` binding. When I make use of the query in my screen, instead of just passing the value straight from the multi-select picker field value, I use the following JS:

```js
return $("Form.Fields.brands").join('|')
```

<img width="876" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/240ae772-2e71-4790-a085-713b06472b3d">

Then I changed the delimiter in the SQL query to `|` and it all works perfectly.

Hope this helps - if you have any questions let me know!

**Comment by Marcello1099** [comment]
I will mark this as solved. Also, there is an alternative to use {{ stringify [field] }}. This will return a JSON array as a string, which then in PostgreSQL can be converted to an array. However, the provided solution also satisfies the problem stated. Thank you!
